tests/unit/misc/userscripts/test_qute_lastpass.py

```diff
@@ -0,0 +1,353 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2014-2020 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for misc.userscripts.qute-lastpass."""
+
+import os
+import json
+import subprocess
+from importlib.machinery import SourceFileLoader
+from importlib.util import spec_from_loader, module_from_spec
+from unittest.mock import MagicMock, call
+
+# qute-lastpass violates naming convention and does not have .py extension
+dirname = os.path.dirname(__file__)
+script_path = os.path.join(
+    dirname, "../../../../misc/userscripts/qute-lastpass")
+spec = spec_from_loader("qute_lastpass", SourceFileLoader(
+    "qute_lastpass",
+    script_path))
+qute_lastpass = module_from_spec(spec)
+spec.loader.exec_module(qute_lastpass)
+
+default_lpass_match = [
+    {
+        "id": "12345",
+        "name": "www.example.com",
+        "username": "fake@fake.com",
+        "password": "foobar",
+        "url": "https://www.example.com",
+    }
+]
+
+
+def get_response_mock(stdout='', stderr=''):
+    response = MagicMock()
+    response.stdout = stdout.encode()
+    response.stderr = stderr.encode()
+
+    return response
+
+
+def setup_subprocess_mock(mocker, stdout='', stderr=''):
+    mocker.patch('subprocess.run')
+
+    subprocess.run.return_value = get_response_mock(stdout, stderr)
```

Comment:
This could be a [pytest fixture](https://docs.pytest.org/en/stable/fixture.html) which returns the patched mock. Then you won't have to pass the `mocker` fixture in here. The test code would then do something like:

```python3
def test_something(lpass_mock):
    lpass_mock.run.return_value = FakeOutput(stdout, stderr)
    ...
```